### PR TITLE
Fail AppInitMain if either disk space check fails

### PR DESCRIPTION
**Pull request by Empact**
Rather than both.

Introduced in 386a6b62a8a1db9dd0f354cb95b7585f555c7e5d, #12653

### CODE REVIEW

**Comment by jimpo** [comment]
utACK 11e85ca

File: src/init.cpp [hunk 1]
```diff
@@ -1631,7 +1631,7 @@ bool AppInitMain(InitInterfaces& interfaces)
 
     // ********************************************************* Step 11: import blocks
 
-    if (!CheckDiskSpace() && !CheckDiskSpace(0, true))
+    if (!CheckDiskSpace() || !CheckDiskSpace(0, true))
```
[the remark this answers is not in the document]
Comment (Empact) [reply src/init.cpp]: Done. #15118 clarifies these more so I'm partly deferring to that.

File: src/init.cpp [hunk 2]
```diff
@@ -1631,7 +1631,7 @@ bool AppInitMain(InitInterfaces& interfaces)
 
     // ********************************************************* Step 11: import blocks
 
-    if (!CheckDiskSpace() && !CheckDiskSpace(0, true))
+    if (!CheckDiskSpace() || !CheckDiskSpace(/* additional_bytes */ 0, /* blocks_dir */ true))
```
Comment (laanwj) [inline src/init.cpp]: When you are changing this anyway: please provide explicit arguments to the first call too, instead of relying on default arguments

**Comment by jamesob** [comment]
Agree with @promag: while we're in the neighborhood, we should specify which disk is full.

**Comment by Empact** [comment]
Updated to address feedback.

**Comment by promag** [comment]
utACK ba8c8b22, thanks!

**Comment by laanwj** [comment]
utACK ba8c8b22272ad40fe2de465d7e745532bab48d3b

**Comment by MarcoFalke** [comment]
Could add a test, if possible somehow?

**Comment by Empact** [comment]
I don't know of any way to set up the scenario - could refactor for more testability with stubbing/mocking.

**Pull request merged**